src/ol/interaction/Select.js

```diff
@@ -255,11 +265,13 @@ class Select extends Interaction {
      */
     this.featureLayerAssociation_ = {};
 
-    const features = this.featureOverlay_.getSource().getFeaturesCollection();
-    listen(features, CollectionEventType.ADD,
-      this.addFeature_, this);
-    listen(features, CollectionEventType.REMOVE,
-      this.removeFeature_, this);
+    if (!this.userSpecifiedFeatureOverlay_) {
```

Comment:
Instead of treating a user layer differently, you could unconditionally listen to the source's `ADDFEATURE` and `REMOVEFEATURE` events:
 ```js
const source = this.featureOverlay_.getSource();
listen(source, VectorEventType.ADDFEATURE, this.addFeature_, this);
listen(source, VectorEventType.REMOVEFEATURE, this.removeFeature_, this);
```

Reply:
Hi! I added fixups for the requested changes. The first three work great. Unfortunately, this one doesn't so well. The changes for this are in a5f4690. Instead of behavior like shown in the GIF in the description, with this I see:

![2018-08-14 21 14 08](https://user-images.githubusercontent.com/1430060/44126560-0b355e20-a007-11e8-9916-b5a2270ac371.gif)

Something about skipping the features makes them not render, which is why I had excluded these when a user layer was specified. If you can suggest any other workarounds I'll be happy to try them out too.